instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -186,14 +193,18 @@ private Context doStart(Context parentContext, REQUEST request, @Nullable Instan
 
     boolean localRoot = LocalRootSpan.isLocalRoot(context);
 
-    spanBuilder.setAllAttributes(attributes);
     Span span = spanBuilder.setParent(context).startSpan();
```

Comment:
I think we need to set attributes here so that samplers and (onStart) span processors can access them

Reply:
Yeah, I think a v2 interface for this should take in account all the possible use cases - in my mind it would just put attributes directly on the span (through the Attributes/AttributesBuilder interfaces if desired) and any odd bits of logic (like the some of metric creators) can filter out or copy specific attributes/sets if they care to.  I'm going to close this PR as outdated and not-fully-thought-through.